Classes/Controller/BackendServiceController.php

```diff
@@ -357,24 +360,13 @@ public function changeBaseWorkspaceAction(string $targetWorkspaceName, NodeInter
             $this->feedbackCollection->add($updateWorkspaceInfo);
 
             // Construct base workspace context
-            $originalContext = $documentNode->getContext();
             $contextProperties = $documentNode->getContext()->getProperties();
             $contextProperties['workspaceName'] = $targetWorkspaceName;
             $contentContext = $this->contextFactory->create($contextProperties);
 
             // If current document node doesn't exist in the base workspace, traverse its parents to find the one that exists
-            $redirectNode = $documentNode;
-            while (true) {
-                $redirectNodeInBaseWorkspace = $contentContext->getNodeByIdentifier($redirectNode->getIdentifier());
-                if ($redirectNodeInBaseWorkspace) {
-                    break;
-                } else {
-                    $redirectNode = $redirectNode->getParent();
-                    if (!$redirectNode) {
-                        throw new Exception(sprintf('Wasn\'t able to locate any valid node in rootline of node %s in the workspace %s.', $documentNode->getContextPath(), $targetWorkspaceName), 1458814469);
-                    }
-                }
-            }
+            $nodesOnPath = $documentNode->getContext()->getNodesOnPath($sitePath, $originalNodePath);
+            $redirectNode = $nodesOnPath[count($nodesOnPath) - 1];
```

Comment:
```suggestion
            $redirectNode = array_pop($nodesOnPath);
```
Then you might get `null` as result and then we could instead use the site node of the context as redirectNode

Reply:
Like this?
`$redirectNode = array_pop($nodesOnPath) ?? $documentNode->getContext()->getRootNode;`

Reply:
`$documentNode->getContext()->getCurrentSiteNode()` should work.
The root node is not loadable in the ui.